src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -240,7 +240,9 @@ private void AddNameAndTypeArgumentsOrParameters(INamedTypeSymbol symbol)
             }
             else if (symbol.IsTupleType)
             {
-                if (CanUseTupleTypeName(symbol))
+                // If top level tuple uses non-default names, there is no way to preserve them
+                // unless we use tuple syntax for the type. So, we give them priority.
+                if (HasNonDefaultTupleElementNames(symbol) || CanUseTupleTypeName(symbol))
```

Comment:
Should we consider a textual representation of the underlying type with names too? Maybe `System.ValueTuple<int a, int b>`?